app/controllers/DraftMetadataChecksResultsController.scala

```diff
@@ -129,28 +128,33 @@ class DraftMetadataChecksResultsController @Inject() (
     }
   }
 
-  def downloadErrorReport(consignmentId: UUID): Action[AnyContent] = standardUserAndTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
-    for {
-      reference <- consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken)
-      errorReport <- draftMetadataService.getErrorReport(consignmentId)
-    } yield {
-      val errorList = errorReport.fileError match {
-        case SCHEMA_VALIDATION =>
-          errorReport.validationErrors.flatMap { validationErrors =>
-            val data = validationErrors.data.map(metadata => metadata.name -> metadata.value).toMap
-            validationErrors.errors.map(error => List(data(filePath), error.property, data(error.property), error.message))
-          }
-        case _ => Nil
+  def downloadErrorReport(consignmentId: UUID, priorityErrorType: FileError = ROW_VALIDATION): Action[AnyContent] = standardUserAndTypeAction(consignmentId) {
+    implicit request: Request[AnyContent] =>
+      for {
+        reference <- consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken)
+        errorReport <- draftMetadataService.getErrorReport(consignmentId)
+      } yield {
+        val orderingPrioritisingRowErrors: Ordering[Error] =
```

Comment:
Should rename this val to something less specific given it might not always be row errors with priority